src/edu/csus/ecs/pc2/clics/API202306/CLICSTeam.java

```diff
@@ -0,0 +1,246 @@
+        Account [] newaccounts = null;
+        Log log = StaticLog.getLog();
+
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            newaccounts = createTeamsFromTeamsJSON(contest, mapper.readValue(jsonfile, CLICSTeam[].class), site, institutionsMap, log);
+        } catch (Exception e) {
+            log.log(Log.WARNING, "could not deserialize team file " + jsonfile, e);
+        }
+        return(newaccounts);
+     }
+
+    /**
+     * Create account list from a strings contains a teams json file
+     *
+     * @param contest the contest (needed for groups)
+     * @param json json to deserialize
+     * @param site the site to create the accounts for
+     * @param institutionsMap hashmap mapping organization id to organization fields (to be removed once organizations
+     *     are part of the contest model like groups).  May be null if not interested in organizations.
+     * @return array of accounts to add, or null on error
+     */
+    public static Account [] fromJSON(IInternalContest contest, String json, int site, HashMap<String,String[]> institutionsMap) {
+        Account [] newaccounts = null;
+        Log log = StaticLog.getLog();
+
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            newaccounts = createTeamsFromTeamsJSON(contest, mapper.readValue(json, CLICSTeam[].class), site, institutionsMap, log);
+        } catch (Exception e) {
+            log.log(Log.WARNING, "could not deserialize team string", e);
+        }
+        return(newaccounts);
+     }
+
+    /**
+     * Converts CLICS teams into a PC2 Account array.
+     *
+     * @param contest The contest
+     * @param teams array of CLICS teams
+     * @param site The site to create accounts on
+     * @param log For errors
+     * @return an Account array of teams converted from the CLICS teams or null if error
+     */
+    private static Account [] createTeamsFromTeamsJSON(IInternalContest contest, CLICSTeam [] teams, int site, HashMap<String,String[]> institutionsMap, Log log) {
+
+        Account [] newaccounts = null;
+        ArrayList<Account> accounts = new ArrayList<Account>();
+        Account account;
+        JSONTool jsontool = new JSONTool(contest, null);
+        PermissionList teamPermissionList = new PermissionGroup().getPermissionList (ClientType.Type.TEAM);
+        boolean error = false;
+
+        // convert each team to an account
+        for(CLICSTeam team: teams) {
+            int teamnum;
+            try {
+                if(StringUtilities.isEmpty(team.label)) {
+                    teamnum = Integer.parseInt(team.id);
+                } else {
+                    teamnum = Integer.parseInt(team.label);
+                }
+            } catch(Exception e) {
+                // Some sort of conversion error - log it and abort
+                log.log(Log.SEVERE, "unable to get team number from label or id", e);
```

Comment:
Suggest adding "CLICS" in front of "label" and "id".

Reply:
Added in next push.